src/components/app/data/hooks/useCanUpgradeWithLearnerCredit.js

```diff
@@ -26,6 +26,9 @@ export default function useCanUpgradeWithLearnerCredit(courseRunKey, queryOption
       const redeemableCourseRun = data.filter((canRedeemData) => (
         canRedeemData.canRedeem && canRedeemData.redeemableSubsidyAccessPolicy
       ))[0];
+
+      // TODO: Consolidate redemption and upgrade data-structure into one.
+      // See usage in components/dashboard/main-content/course-enrollments/data/hooks.js -> useCourseUpgradeData()
```

Comment:
Could you expand on a bit on what this means more concretely?

Reply:
Since the feedback received is pointing in the direction to consolidate the logic at the source now (vs the wonky transformation within `getSubsidyToApplyForCourse`), this comment is now moot :)